common/display/displayqueue.cpp

```diff
@@ -399,6 +401,10 @@ bool DisplayQueue::QueueUpdate(std::vector<HwcLayer*>& source_layers,
   bool composition_passed = true;
   bool disable_ovelays = state_ & kDisableOverlayUsage;
 
+  //FIXME? disalbe overlay when need apply CTM but is not supported in
+  // display so need to fallback to GPU
+  disable_ovelays = disable_ovelays | ( state_ & kApplyGPUColorMatrix);
```

Comment:
Can we avoid this ?

Reply:
@kalyankondapally  But how can we make use it goes to the GPU composition when CTM is not supported directly in display

Reply:
Once Validate layers is done, we can set force gpu rendering for each display plane state:
https://github.com/intel/IA-Hardware-Composer/blob/master/common/display/displayplanestate.h#L157

Once that is done, we need to ensure each plane has offscreen target i.e. 
Check if  NativeSurface* surface = last_plane.GetOffScreenTarget(); is available, if this is null ensure offscreen target is set by callng:
https://github.com/intel/IA-Hardware-Composer/blob/master/common/display/displayqueue.cpp#L241